src/Builder/CorrelationIdsRegistryBuilder.php

```diff
@@ -56,7 +56,7 @@ public function buildFromRequestHeaders(array $requestHeaders): CorrelationIdsRe
         return $this->build(
             $this->extractRequestHeader(
                 $sanitizedRequestHeaders,
-                $this->sanitize($this->provider->provideParentCorrelationIdHeaderName())
+                $this->sanitize($this->provider->provideCurrentCorrelationIdHeaderName())
```

Comment:
Have you checked that the tests are not failing? Also if this case is not covered, can you extend the unit tests?

Reply:
the tests were actually failing with that change :)